tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/paging/PagingSendTest.java

```diff
@@ -288,6 +296,408 @@ public void testPagingDoesNotDuplicateBatchMessagesAfterPagingStarted() throws E
+         assertTrue(waitForMessages(queue, totalMessages, 10000));
+
+         PagingStore queuePagingStore = queue.getPagingStore();
+         assertTrue(queuePagingStore != null && queuePagingStore.isPaging());
+
+         long existingPages = queuePagingStore.getNumberOfPages();
+         assertTrue(existingPages > 4);
+
+         // restart the server with a smaller pageLimitSize < existing pages.
+         server.stop(true);
+         waitForServerToStop(server);
+
+         addressSettings.setPageLimitBytes((long) (size * 4));
+         server.getAddressSettingsRepository().addMatch(queueName, addressSettings);
+
+         // server will start regardless of current page count > (pageLimitBytes / pageSizeBytes)
+         try {
+            server.start();
+            waitForServerToStart(server);
+
+            // verify current situation
+            queue = server.locateQueue(queueAddr);
+
+            assertTrue(waitForMessages(queue, totalMessages, 10000));
+
+            queuePagingStore = queue.getPagingStore();
+            assertTrue(queuePagingStore != null && queuePagingStore.isPaging() && queuePagingStore.isPageFull());
+
+            long currentPages = queuePagingStore.getNumberOfPages();
+            assertEquals(existingPages, currentPages);
+
+            long maxPages = queuePagingStore.getPageLimitBytes() / queuePagingStore.getPageSizeBytes();
+            assertTrue(currentPages > maxPages);
+
+            //consume messages until current pages goes down to below maxPage
+            locator = createFactory(isNetty());
+            final int numMessages = 25;
+            try (ClientSessionFactory csf = createSessionFactory(locator)) {
+               session = csf.createSession(false, true);
+               session.start();
+               ClientConsumer consumer = session.createConsumer(queueName);
+               for (int i = 0; i < numMessages; i++) {
+                  ClientMessage message = consumer.receive(5000);
+                  assertNotNull(message);
+                  message.acknowledge();
+                  session.commit();
+               }
+
+               currentPages = queuePagingStore.getNumberOfPages();
+               assertTrue(currentPages < maxPages);
+               // check page store not page full
+               assertTrue(queuePagingStore.isPaging() && !queuePagingStore.isPageFull());
+
+               //send messages one by one until page full
+               ClientProducer producer = session.createProducer(queueName);
+               boolean isFull = false;
+               try {
+                  for (int i = 0; i < numMessages; i++) {
+                     ClientMessage message = createMessage(session, messageSize, i, null);
+                     producer.send(message);
+                     session.commit();
+                  }
+               } catch (ActiveMQAddressFullException e) {
+                  isFull = true;
+                  session.close();
```

Comment:
session close should move after the try-catch as we always want to do it, and it would otherwise fail to happen if the loop somehow succeeded.

EDIT: actually, since the wider try-catch with the ClientSessionFactory csf is there, that means it always closes right...so can this just be omitted rather than look wrong?